### PR TITLE
[fix][auto-recovery] Fix metadata store deadlock due to BookkeeperInternalCallbacks.Processor

### PR DESCRIPTION
**Pull request by horizonzy**
### Motivation
The Auditor will start AuditorCheckAllLedgersTask to checkAllLedgers. It will iterate each ledger using `ledgerManager.asyncProcessLedgers`, then callback checkLedgersProcessor in the metadata store thread.

https://github.com/apache/bookkeeper/blob/e8da8eb6cb7c8ef52336a51d4f7348ae07986c26/bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/AuditorCheckAllLedgersTask.java#L158-L248

In the checkLedgersProcessor, it will invoke `ledgerUnderreplicationManager.isLedgerReplicationEnabled()`, it's a metadata store sync operation, so deadlock.

There is the stack info:
```
"metadata-store-40-1" #161 prio=5 os_prio=31 cpu=43.17ms elapsed=24.19s tid=0x000000015932b000 nid=0x1720b waiting on condition  [0x00000004d2be9000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@17.0.3.1/Native Method)
	- parking to wait for  <0x0000200002af57a8> (a java.util.concurrent.CompletableFuture$Signaller)
	at java.util.concurrent.locks.LockSupport.parkNanos(java.base@17.0.3.1/LockSupport.java:252)
	at java.util.concurrent.CompletableFuture$Signaller.block(java.base@17.0.3.1/CompletableFuture.java:1866)
	at java.util.concurrent.ForkJoinPool.unmanagedBlock(java.base@17.0.3.1/ForkJoinPool.java:3463)
	at java.util.concurrent.ForkJoinPool.managedBlock(java.base@17.0.3.1/ForkJoinPool.java:3434)
	at java.util.concurrent.CompletableFuture.timedGet(java.base@17.0.3.1/CompletableFuture.java:1939)
	at java.util.concurrent.CompletableFuture.get(java.base@17.0.3.1/CompletableFuture.java:2095)
	at org.apache.pulsar.metadata.bookkeeper.PulsarLedgerUnderreplicationManager.isLedgerReplicationEnabled(PulsarLedgerUnderreplicationManager.java:735)
	at org.apache.bookkeeper.replication.AuditorCheckAllLedgersTask.lambda$checkAllLedgers$2(AuditorCheckAllLedgersTask.java:167)
	at org.apache.bookkeeper.replication.AuditorCheckAllLedgersTask$$Lambda$375/0x0000000800fd5340.process(Unknown Source)
	at org.apache.pulsar.metadata.bookkeeper.AbstractHierarchicalLedgerManager.lambda$asyncProcessLedgersInSingleNode$2(AbstractHierarchicalLedgerManager.java:211)
	at org.apache.pulsar.metadata.bookkeeper.AbstractHierarchicalLedgerManager$$Lambda$410/0x0000000800ffe9e8.accept(Unknown Source)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(java.base@17.0.3.1/CompletableFuture.java:718)
	at java.util.concurrent.CompletableFuture.postComplete(java.base@17.0.3.1/CompletableFuture.java:510)
	at java.util.concurrent.CompletableFuture.complete(java.base@17.0.3.1/CompletableFuture.java:2147)
	at org.apache.pulsar.metadata.impl.ZKMetadataStore.handleGetChildrenResult(ZKMetadataStore.java:291)
	at org.apache.pulsar.metadata.impl.ZKMetadataStore.lambda$batchOperation$5(ZKMetadataStore.java:222)
	at org.apache.pulsar.metadata.impl.ZKMetadataStore$$Lambda$296/0x0000000800f859f0.run(Unknown Source)
	at java.util.concurrent.Executors$RunnableAdapter.call(java.base@17.0.3.1/Executors.java:539)
	at java.util.concurrent.FutureTask.run(java.base@17.0.3.1/FutureTask.java:264)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(java.base@17.0.3.1/ScheduledThreadPoolExecutor.java:304)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@17.0.3.1/ThreadPoolExecutor.java:1136)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@17.0.3.1/ThreadPoolExecutor.java:635)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(java.base@17.0.3.1/Thread.java:833)
```

### Modifications
Refer to bookkeeper https://github.com/apache/bookkeeper/blob/e8da8eb6cb7c8ef52336a51d4f7348ae07986c26/bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/AbstractZkLedgerManager.java#L607-L611.

Using scheduler to trigger the processor.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

**Review by hangc0276 — APPROVED** [review]
Good job!

**Review by mattisonchao — APPROVED** [review]
👍🏼  Nice catch!

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/AbstractHierarchicalLedgerManager.java [hunk 1]
```diff
@@ -206,10 +206,11 @@ protected void asyncProcessLedgersInSingleNode(
                             mcb = new BookkeeperInternalCallbacks.MultiCallback(activeLedgers.size(), finalCb, ctx,
                             successRc, failureRc);
                     // start loop over all ledgers
-                    for (Long ledger : activeLedgers) {
-                        processor.process(ledger, mcb);
-                    }
-
+                    scheduler.submit(() -> {
```
[the remark this answers is not in the document]
Comment (horizonzy) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/AbstractHierarchicalLedgerManager.java:209]: I think making `processor.process` in the scheduler is good enough, the processor is not under-controlled. If there are no activeLedgers, no need to switch thread.

[the remark this answers is not in the document]
Comment (mattisonchao) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/AbstractHierarchicalLedgerManager.java:209]: Okay.

**Review by zymap — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Review by poorbarcode — APPROVED** [review]

**Pull request merged**